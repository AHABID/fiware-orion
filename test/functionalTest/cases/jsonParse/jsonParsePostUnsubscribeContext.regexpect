1: ***********************************
HTTP/1.1 200 OK
Content-Length: 206
Content-Type: application/json
Date: REGEX(.*)

{
    "statusCode": {
        "code": "404",
        "details": "subscriptionId: '012345678901234567890123'",
        "reasonPhrase": "No context element found"
    },
    "subscriptionId": "012345678901234567890123"
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 134
Content-Type: application/json
Date: REGEX(.*)

{
    "subscribeResponse": {
        "duration": "P5Y",
        "subscriptionId": "REGEX([0-9a-f]{24})",
        "throttling": "P5Y"
    }
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 122
Content-Type: application/json
Date: REGEX(.*)

{
    "statusCode": {
        "code": "200",
        "reasonPhrase": "OK"
    },
    "subscriptionId": "REGEX([0-9a-f]{24})"
}
4: ***********************************
