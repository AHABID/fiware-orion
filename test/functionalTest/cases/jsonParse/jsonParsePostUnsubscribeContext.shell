echo "1: ***********************************"
url="/NGSI10/unsubscribeContext"
payload='{
  "subscriptionId": "012345678901234567890123"
}'
curlJson ${url} "${payload}"

echo "2: ***********************************"
url="/NGSI10/subscribeContext"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "occupancy",
    "lightstatus"
  ],
  "reference": "127.0.0.1",
  "duration": "P5Y",
  "restriction": {
    "attributeExpression": "testRestriction",
    "scopes" : [
      {
        "type" : "t1",
        "value" : "v1"
      },
      {
        "type" : "t2",
        "value" : "v2"
      }
    ]
  },
  "notifyConditions": [
    {
      "type": "ONCHANGE",
      "condValues": [
        "CondValue1",
        "CondValue2"
      ]
      ,
      "restriction": "restriction"
    }
  ],
  "throttling": "P5Y"
}'
curlJson ${url} "${payload}"

subscriptionId=$(echo "$response" | grep subscriptionId | awk -F\" '{ print $4 }')

echo "3: ***********************************"
url="/NGSI10/unsubscribeContext"
payload='{
  "subscriptionId": "'${subscriptionId}'"
}'
curlJson ${url} "${payload}"

echo "4: ***********************************"
