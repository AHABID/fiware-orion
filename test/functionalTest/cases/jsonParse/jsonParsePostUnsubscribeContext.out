1: ***********************************
HTTP/1.1 200 OK
Content-Length: 206
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:20:29 GMT

{
    "statusCode": {
        "code": "404", 
        "details": "subscriptionId: '012345678901234567890123'", 
        "reasonPhrase": "No context element found"
    }, 
    "subscriptionId": "012345678901234567890123"
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 134
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:20:29 GMT

{
    "subscribeResponse": {
        "duration": "P5Y", 
        "subscriptionId": "538c33cd947d6969da9cb6e6", 
        "throttling": "P5Y"
    }
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 122
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:20:29 GMT

{
    "statusCode": {
        "code": "200", 
        "reasonPhrase": "OK"
    }, 
    "subscriptionId": "538c33cd947d6969da9cb6e6"
}
4: ***********************************
